internal/httpproxy/http_proxy.go

```diff
@@ -380,6 +384,7 @@ func NewProxy(cfg Config) (*Proxy, error) {
 		port:                cfg.Port,
 		connectValidator:    cfg.ConnectValidator,
 		k8sAPIServerToken:   cfg.K8sAPIServerToken,
+		config:              cfg,
```

Comment:
I think it's nicer that the `Proxy` keeps a `config` field than flatten the config fields. I could remove the `Proxy.port`, `Proxy.connectValidator`, and `Proxy.k8sAPIServerToken` fields in another PR if we decide to go with this `config` field.

Reply:
Makes sense to me